pkg/graveler/graveler.go

```diff
@@ -1571,6 +1571,29 @@ func (g *KVGraveler) listStagingArea(ctx context.Context, b *Branch) (ValueItera
 	return NewCombinedIterator(itrs...), nil
 }
 
+func (g *KVGraveler) listSealedTokens(ctx context.Context, b *Branch) (ValueIterator, error) {
+	var itrs []ValueIterator
+
+	for _, st := range b.SealedTokens {
+		it, err := g.StagingManager.List(ctx, st, 1)
+		if err != nil {
+			for _, it = range itrs {
+				it.Close()
+			}
+			return nil, err
+		}
+		itrs = append(itrs, it)
+	}
+
+	if len(itrs) == 0 {
+		return nil, nil
+	}
```

Comment:
1. We can check it before the 'for' loop.
2. Would expect to get an iterator with no values and not nil if no error was found

Reply:
changed this function